docs/enhancements/infrastructure-monitoring-service-api/lifecycle.md

```diff
@@ -0,0 +1,530 @@
+    apiVersion: monitoring.coreos.com/v1
+    kind: PrometheusRule
+    metadata:
+      name: node-exporter-rules
+      namespace: openshift-monitoring
+    spec:
+      groups:
+        - name: node-exporter
+          rules:
+            - alert: NodeClockNotSynchronising  #### (alarm_definitions.alarm_name)
+              annotations:
+                description: Clock at {{ $labels.instance }} is not synchronising. Ensure NTP is configured on this host.
+                runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/cluster-monitoring-operator/NodeClockNotSynchronising.md #### (alarm_definitions.proposed_repair_actions)
+                summary: Clock not synchronising. #### (alarm_definitions.alarm_description)
+              expr: |
+                min_over_time(node_timex_sync_status{job="node-exporter"}[5m]) == 0
+                and
+                node_timex_maxerror_seconds{job="node-exporter"} >= 16
+              for: 10m
+              labels: 
+                severity: critical #### (alarm_definitions.extensions)
+    ```
+
+
+
+## Tables during ready and Alertmanager sent a notification
+
+- Interacting with two tables `alarm_event_record` and `alarm_subscription_info`
+
+### For a given ResourceTypeID and AlarmName (coming from AM alert), find the AlarmDefinitionID and ProbableCauseID
+- Find resourceType ID and alart name
+  - Get managedcluster-id and alert name from alertmanager alerts (e.g NodeClockNotSynchronising2)
+  - Ask inventory for ResourceType ID using managedcluster-id (e.g b3e7149e-d471-4d0f-aaa6-d5e9aa9e713a)
+    ```go
+    // should be something we compute 
+    func getResourceTypeID(managedClusterId uuid.UUID, class ResourceClass, resourceType ResourceType) uuid.UUID {
+      return v5UUID(fmt.Sprintf("%s-%s-%s", managedClusterId.String(), class, resourceType))
+    }
+    ```
+- Run following. `resource_type_id` should be enough to uniquely identify the corresponding alarm dictionary
+  and its associated alarm definitions and finally use `alarm_name` to filter out the exact rule.
+    ```sql
+    SELECT
+        ad.alarm_definition_id,
+        pc.probable_cause_id
+    FROM
+        alarm_definitions ad
+        JOIN
+          alarm_dictionary adict
+          ON ad.alarm_last_change = adict.alarm_dictionary_version
+        JOIN
+          probable_causes pc
+          ON ad.alarm_definition_id = pc.alarm_definition_id
+    WHERE
+        adict.resource_type_id = '481688c8-2782-4534-a9de-88ca5154411d'
+        AND ad.alarm_name = 'LowMemory';
+    ```
+
+
+### Notification tracking
+
+- Collect all subscriber info including ID, callback and filter
+- Collect all the AlarmEventRecord based on sequence number and optionally the filter. Here we are collecting everything that's "CRITICAL"
+    ```sql
+    SELECT aer.*
+    FROM alarm_event_record aer
+             JOIN alarm_subscription_info asi ON asi.alarm_subscription_id = 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'
```

Comment:
We should loop though the alarm_subscription_info table?

Reply:
Good point. We can do one "full" query instead of looping and also using `*` is a bad idea. This is query is just show that it's possible to grab only the latest events for each subscriber...and definitely not optimized.